tmt/utils.py

```diff
@@ -1063,6 +1074,28 @@ def validate_fmf_id(fmf_id):
     return (True, '')
 
 
+def generate_runs(path, id_):
+    """ Generate absolute paths to runs from path """
+    # Prepare absolute workdir path if --id was used
+    if id_:
+        if '/' not in id_:
+            id_ = os.path.join(path, id_)
+        if os.path.isabs(id_):
+            if os.path.exists(id_):
+                yield id_
+            return
+    if not os.path.exists(path):
+        return
+    for filename in os.listdir(path):
+        abs_path = os.path.join(path, filename)
+        invalid_id = id_ and abs_path != id_
```

Comment:
I'm not sure about the `id_ and abs_path != id_` condition. Could you please explain?

Reply:
This was basically copied over from `Status` and it probably deserves a comment in the code. The reasoning is: `id_` can be `None` (no filtering based on id). In this case, `abs_path` is "valid" (it matches the id it is filtered with). However when `id_` is not None, `abs_path` is valid only if it is equal to the specified `id_` (after conversion to absolute path).

Reply:
Thanks for the clarification.